ablySpec/RealtimeClientChannel.swift

```diff
@@ -181,6 +181,25 @@ class RealtimeClientChannel: QuickSpec {
                     expect(channel.state).toEventually(equal(ARTRealtimeChannelState.Failed), timeout: testTimeout)
                 }
 
+                // RTL4f
+                pending("should transition the channel state to FAILED if ATTACHED ProtocolMessage is not received") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    let client = ARTRealtime(options: options)
+                    client.setTransportClass(TestProxyTransport.self)
+                    client.connect()
+                    defer { client.close() }
+
+                    expect(client.connection().state).toEventually(equal(ARTRealtimeConnectionState.Connected), timeout: testTimeout)
+                    let transport = client.transport as! TestProxyTransport
+                    transport.actionsIgnored += [.Attached]
+
+                    let channel = client.channel("test")
+                    channel.attach()
```

Comment:
@mattheworiordan `realtimeRequestTimeout` is currently readonly and I can't hack that method because it is a static method. The user should have access to change that option or not?